### ISSUE TITLE
System.AccessViolationException -- Loading tensorflow model

### ISSUE THREAD

**Issue by vera-dania** [dotnet/machinelearning#4176]
### System information

- **Windows 10 Pro for Workstations - 1809**:
- **3.0.100-preview3-010431**: 

### Issue

- **What did you do?**
I am loading a tensorflow model using ML.NET Model.LoadTensorFlowModel method.
Received this exception:

- **What happened?**
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt. I am unable to locate the source of this exception. I only receive this error when running the application in debug mode.

- **What did you expect?**
My model to successfully load and be able to make predictions.

### Source code / logs

      var pipline = context.Transforms.Conversion.MapValueToKey("LabelKey", "Label")
        .Append(context.Transforms.LoadImages("input", "images", nameof(ImageData.ImagePath)))
        .Append(context.Transforms.ResizeImages("input", GenderSettings.ImageHeight, GenderSettings.ImageWidth, "input"))
        .Append(context.Model.LoadTensorFlowModel("./modelGender/tensorflow_gender__graph.pb")
          .ScoreTensorFlowModel(new[] {"cross_entropy"}, new[] {"input"}, addBatchDimensionInput: true))
        .Append(context.Transforms.Conversion.MapKeyToValue("PredictedLabelValue", "PredictedLabel"))
        .AppendCacheCheckpoint(context);

It is not showing any stacktrace or log.

**Comment by colbylwilliams** [comment]
@codemzs @KsenijaS These two open issues in the samples repo may be related.  Note one of them is ONNX, the other is Tensorflow:
https://github.com/dotnet/machinelearning-samples/issues/655
https://github.com/dotnet/machinelearning-samples/issues/629
I'm unable to repro as it seems environmental, but I have a hunch (guess) that it may be a x86 vs. x64 vs. "Any CPU" defaulting to the wrong arch.. thing.

**Comment by Anki7s** [comment]
> 
> 
> @codemzs @KsenijaS These two open issues in the samples repo may be related. Note one of them is ONNX, the other is Tensorflow:
> 
>     * [dotnet/machinelearning-samples#655](https://github.com/dotnet/machinelearning-samples/issues/655)
> 
>     * [dotnet/machinelearning-samples#629](https://github.com/dotnet/machinelearning-samples/issues/629)
> 
> 
> I'm unable to repro as it seems environmental, but I have a hunch (guess) that it may be a x86 vs. x64 vs. "Any CPU" defaulting to the wrong arch.. thing.

@colbylwilliams  I'm using windows 7 with Microsoft Visual Studio Community 2019
Version 16.2.5
VisualStudio.16.Release/16.2.5+29306.81
Microsoft .NET Framework
Version 4.7.03062

Tried to execute it on x64 but still facing this problem.

**Comment by sergioprates** [comment]
This error occurs with me when the input model variable name is incorrect, upload your model at this site https://lutzroeder.github.io/netron/ and check if the output variable name is equal at your ML.NET import

**Comment by KsenijaS** [comment]
@vera-dania Can you please share tensorflow model that you used

**Comment by KsenijaS** [comment]
Closing this issue for now, feel free to reopen it when you have provided the model

**Comment by vera-dania** [comment]
@KsenijaS Should I share the model here?

**Comment by KsenijaS** [comment]
Yes

**Comment by vera-dania** [comment]
I can't upload the model because of the size. Only file smaller than 10MB are allowed

**Comment by KsenijaS** [comment]
Can you upload it to some public share like Google drive?

**Comment by Abdiflame** [comment]
I was facing the same problem.

**Reason:**
The TensorFlow version installed on the machine was different from the TensorFlow version that I used to build the model.

**Solution:**
Uninstall TensorFlow and install the same version that matches your model.

Hope it helps 👍
